### PR TITLE
use postgresql as database for gateway and repository-server

### LINKED ISSUE #315: Configure API projects to use Postgres
### What is the expected enhancement?

To simplify the process for `repository` and for `gateway` we started the MVP's using SQLite. Preparing the projects for production we would like to configure Postgres as the database to use.

### PR DESCRIPTION
**Pull request by akihikokuroda**
### Summary
Fix #315

### CODE REVIEW

**Review by IceKhan13 — APPROVED** [review]
LGTM

@Tansito be aware of this in your PR https://github.com/Qiskit-Extensions/quantum-serverless/pull/447

**Review by pacomf — APPROVED** [review]

File: gateway/main/settings.py [hunk 1]
```diff
@@ -89,17 +92,40 @@
...
+    "version": 1,
+    "disable_existing_loggers": False,
+    "handlers": {
+        "console": {
+            "class": "logging.StreamHandler",
+        },
+    },
+    "root": {
+        "handlers": ["console"],
+        "level": "DEBUG",
```
Comment (Tansito) [inline gateway/main/settings.py:105]: We can open an issue and do it later but we should be able to modify the level of the LOGs through an environment variable, for example.

File: gateway/main/settings.py [hunk 2]
```diff
@@ -89,17 +92,40 @@
...
+        "PORT": os.environ.get("DATABASE_PORT", "5432"),
+    },
+    "test": {
         "ENGINE": "django_prometheus.db.backends.sqlite3",
         "NAME": BASE_DIR / "db.sqlite3",
-    }
+    },
 }
 
+if "test" in sys.argv:
```
Comment (Tansito) [inline gateway/main/settings.py:127]: Just as a doubt @akihikokuroda , now if we want to use `sqlite` what command do we need to use: `python manage.py runserver test`?

File: infrastructure/helm/quantumserverless/charts/gateway/templates/deployment.yaml [hunk 3]
```diff
@@ -27,6 +27,10 @@ spec:
       serviceAccountName: {{ include "gateway.serviceAccountName" . }}
       securityContext:
         {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      initContainers:
+        - name: waitpostresql
+          image: actions/pg_isready
+          command: ['sh', '-c', 'until pg_isready -U {{ .Values.database.user | quote }} -d "dbname={{ .Values.database.name }}" -h {{ .Release.Name }}-postgresql -p {{ .Values.database.port | quote }}; do echo waiting for myservice; sleep 2; done']
```
Comment (Tansito) [inline infrastructure/helm/quantumserverless/charts/gateway/templates/deployment.yaml:33]: Correct me if I'm wrong @akihikokuroda , @psschwei but now from #505 we are not using or requiring the use of {{ .Release.Name }}, right?

Comment (Tansito) [reply infrastructure/helm/quantumserverless/charts/gateway/templates/deployment.yaml:33]: And thinking in loud. I would suggest to pass the host entirely as we are doing with the user and so on, I don't know what do you think (thinking that we are going to have the DBs outside the cluster, initially).

Comment (Tansito) [reply infrastructure/helm/quantumserverless/charts/gateway/templates/deployment.yaml:33]: by the way, won't it be required in this case the password?

File: gateway/main/settings.py (hunk 2 above)
Comment (akihikokuroda) [reply gateway/main/settings.py:127]: It doesn't give the choice to use `sqlite` for runtime.  It is for using the sqlite for the unit tests.

File: docker-compose-dev.yml [hunk 4]
```diff
@@ -84,10 +84,16 @@ services:
       - SITE_HOST=http://gateway:8000
 #      - SETTINGS_AUTH_MECHANISM=default #custom_token
 #      - SETTINGS_TOKEN_AUTH_URL=<URL_FOR_TOKEN_VERIFICATION>
```
Comment (Tansito) [inline docker-compose-dev.yml:86]: Just as a last comment @akihikokuroda I don't know if we can remove these lines commented if we are not going to need them.

**Review by Tansito — APPROVED** [review]

File: docker-compose-dev.yml (hunk 4 above)
Comment (akihikokuroda) [reply docker-compose-dev.yml:86]: These lines were added in https://github.com/Qiskit-Extensions/quantum-serverless/pull/524.  I'll leave them there.

**Pull request merged**